ISDeploymentCmd/deploy_project.cs

```diff
@@ -19,10 +22,60 @@ public int DeployProject(string folderName, string projectName, byte[] projectSt
                     cmd.Parameters.Add("@project_name", SqlDbType.NVarChar).Value = projectName;
                     cmd.Parameters.Add("@project_stream", SqlDbType.VarBinary).Value = projectStream;
                     cmd.Parameters.Add("@operation_id", SqlDbType.BigInt).Direction = ParameterDirection.Output;
-                    connection.Open();
+                    if (connection.State!=ConnectionState.Open)
+                    {
+                        connection.Open();
+                    }
                     return Convert.ToInt32(cmd.ExecuteScalar());
                 }
             }
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="folderName"></param>
+        /// <param name="connection"></param>
+        /// <returns></returns>
+        public int CreateFolder(string folderName, SqlConnection connection = null)
```

Comment:
I'm going to refactor CreateFolder into a separate file as another partial class to fit the pattern of my CatalogProcedures library.